src/audio/peak_volume.c

```diff
@@ -0,0 +1,439 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2021 Intel Corporation. All rights reserved.
+//
+// Author: Rander Wang <rander.wang@linux.intel.com>
```

Comment:
@RanderWang can you make this last patch a new PR as I can merge the first patches. We really need to add the peakvolume and fade to the existing volume code. Maybe easier to add peakvol first then the fade with help from @singalsu

Reply:
sure, please check https://github.com/thesofproject/sof/pull/4634. I understand how to implement fade function. it will cost more time and simple volume function is more important, so fade is delayed. It is happy that @singalsu can help it.

Reply:
Thanks @RanderWang, I've asked @singalsu to implement the fade processing to free you up on the IPC4 infrastructure. Can you just have a function that does a simple memcpy() for fade today OR a comment where it should be done in the patch (this will allow @singalsu or @ShriramShastry to work on the fade).

Reply:
sure

Reply:
Yep, we have fade between any gain values in current volume component. I'll see how we utilize it for this.